### PR TITLE
Adding links to Lifecycle of a transaction doc

### PR DESCRIPTION
**Pull request by lsilva01**
This PR adds some links to the code in the "Transaction Lifecycle" document.

### CODE REVIEW

**Comment by glozow** [comment]
Thank you!! :)

File: transaction-lifecycle.md [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+in the next block](https://github.com/bitcoin/bitcoin/blob/1a369f006fd0bec373b95001ed84b480e852f191/src/miner.cpp#L226). This means a "package" consisting of a 5 sat/vB parent and 100 sat/vB child may
+get picked over a 20 sat/vB transaction, but not over a singular 100 sat/vB transaction with no
+dependencies.
+
+### Block Relay
+
+Relevant code: [BIP152](https://github.com/bitcoin/bips/blob/master/bip-0152.mediawiki)
+
+Blocks can contain megabytes worth of transactions, and nodes that keep a mempool have seen
+at least 98% of them already (FIXME source: [gmax's block relay talk](https://www.youtube.com/watch?v=EHIuuKCm53o)).
```
Comment (glozow) [inline transaction-lifecycle.md:298]: amazing! You knew exactly which talk I was talking about :')

File: transaction-lifecycle.md [hunk 2]
```diff
@@ -0,0 +1,363 @@
...
+the [corresponding UTXO](https://github.com/bitcoin/bitcoin/blob/1a369f006fd0bec373b95001ed84b480e852f191/src/validation.cpp#L1469) and passed into the script interpreter. The [script interpreter](https://github.com/bitcoin/bitcoin/blob/1a369f006fd0bec373b95001ed84b480e852f191/src/script/interpreter.cpp#L431) simply
+[evaluates the series of opcodes](https://github.com/bitcoin/bitcoin/blob/1a369f006fd0bec373b95001ed84b480e852f191/src/script/interpreter.cpp#L508) and data based only on the arguments passed to it. One such argument
+is a [set of script verification flags](https://github.com/bitcoin/bitcoin/blob/1a369f006fd0bec373b95001ed84b480e852f191/src/validation.cpp#L922) indicating which rules to apply during script verification.
+
+For example, the [`OP_CHECKSEQUENCEVERIFY`](https://github.com/bitcoin/bitcoin/blob/1a369f006fd0bec373b95001ed84b480e852f191/src/script/script.cpp#L134) opcode [repurposed](https://github.com/bitcoin/bitcoin/pull/7524) `OP_NOP3`. The script verification flag
```
Comment (glozow) [inline transaction-lifecycle.md:152]: Nice job finding the PR! In this case, perhaps BIP 112 would be better? https://github.com/bitcoin/bips/blob/master/bip-0112.mediawiki

**Review by glozow — COMMENTED** [review]
Thanks so much for this PR! I'm very impressed with how many links you've included :) I particularly like how some phrases representing logic, e.g. "may or may not" and "freezes chainstate," are linking to code as well!

A few broadstroke suggestions:
- For BIPs, link to the BIP
- For class names like [`CTransaction`](https://doxygen.bitcoincore.org/class_c_transaction.html) and `CTxMemPool`, I think we should link to doxygen, which is more stable (and perhaps a more user-friendly next step for the reader) than linking to current master

**Comment by glozow** [comment]
Would it be possible for you to do a `git rebase transaction-lifecycle`? Should end up with a diff just adding the links (rather than the entire file)

File: transaction-lifecycle.md (hunk 2 above)
Comment (lsilva01) [reply transaction-lifecycle.md]: Good idea. This BIP also has a link to the PR.

File: transaction-lifecycle.md (hunk 1 above)
Comment (lsilva01) [reply transaction-lifecycle.md:298]: Great. So I got it right. :D

**Comment by lsilva01** [comment]
Thank you for the review.
I changed some links to doxygen and all BIPs mentioned are linked.
The "File changed" tab now shows the diff.

**Pull request merged**